pip/index.py

```diff
@@ -330,13 +335,16 @@ def mkurl_pypi_url(url):
                 ) or req.url_name
 
         if url_name is not None:
-            locations = [
-                mkurl_pypi_url(url)
-                for url in self.index_urls] + self.find_links
-        else:
-            locations = list(self.find_links)
+            return [mkurl_pypi_url(url) for url in self.index_urls]
+        return []
+
+    def _find_all_versions(self, req):
```

Comment:
Yes, that might be the next step, but that would require some changes on _get_pages, _get_page, HtmlPage and I'd prefer to do it in an other PR...